elasticapm/traces.py

```diff
@@ -543,6 +562,22 @@ def to_dict(self) -> dict:
             self.context["tags"] = self.labels
         if self.context:
             self.autofill_resource_context()
+            # otel attributes and spankind need to be top-level
+            if "otel_spankind" in self.context:
+                result["otel"] = {"span_kind": self.context.pop("otel_spankind")}
+            if elasticapm.get_client().check_server_version(gte=(7, 16)):
```

Comment:
On [line 401](https://github.com/elastic/apm-agent-python/pull/1411/files#diff-42a05cd7587872ceaa81c35f1bd163e5420820d2cac99c70a32750230df9c508R401), you're checking first if `get_client()` returns something. Is this not necessary here?

Reply:
It shouldn't be needed in either place in normal operation. I had to add it on 401 to get a couple of transaction_store_tests to pass, like: https://github.com/elastic/apm-agent-python/blob/97188c362497910762e5b1ef8fc53e4011ec75b8/tests/instrumentation/transactions_store_tests.py#L300

It doesn't actually instantiate a Client, just uses the Tracer object directly, so the test was failing. I'll add a little comment on 401.